### PR TITLE
 Fix flaky test issues in SchedulerPoolSuite

### PR DESCRIPTION
**Pull request by huangzhir**
### _Why are the changes needed?_

To fix issue https://github.com/apache/kyuubi/issues/4713, a PR  https://github.com/apache/kyuubi/pull/4714 was submitted, but it had Flaky test issues. After 50 local tests, it succeeded 38 times and failed 12 times.
This PR addresses the issue of flaky tests.

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.readthedocs.io/en/master/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by huangzhir** [comment]
Here are the test results for 50 local tests:

Before the modification:
total_time: 823, avg_time: 16
total: 50, success: 38, fail: 12

After the modification:
total_time: 916, avg_time: 18
total: 50, success: 50, fail: 0.

**Review by pan3793 — APPROVED** [review]

**Review by cxzl25 — APPROVED** [review]

File: externals/kyuubi-spark-sql-engine/src/test/scala/org/apache/kyuubi/engine/spark/SchedulerPoolSuite.scala [hunk 1]
```diff
@@ -101,6 +101,10 @@ class SchedulerPoolSuite extends WithSparkSQLEngine with HiveJDBCTestHelper {
       })
       threads.shutdown()
       threads.awaitTermination(20, SECONDS)
+      // make sure the SparkListener has received the finished events for job1 and job2.
```
Comment (cfmcgrady) [inline externals/kyuubi-spark-sql-engine/src/test/scala/org/apache/kyuubi/engine/spark/SchedulerPoolSuite.scala:104]: if the root cause is the event not being received, let's call `KyuubiSparkContextHelper.waitListenerBus()` to make sure there are no more events in the spark event queue.

Comment (huangzhir) [reply externals/kyuubi-spark-sql-engine/src/test/scala/org/apache/kyuubi/engine/spark/SchedulerPoolSuite.scala:105]: Thank you for pointing out, KyuubiSparkContextHelper.waitListenerBus is indeed a better idea.The average execution time is reduced by 1 second.

total_time: 882 ,avg_time: 17
total: 50, success: 50, fail: 0

**Comment by pan3793** [comment]
Thanks, merged to master/1.7

**Comment by cfmcgrady** [comment]
Good catch! thank you for the fix. @huangzhir

**Pull request closed without merging**